dsc_lib/src/configure/mod.rs

```diff
@@ -152,6 +154,72 @@ impl Configurator {
         Ok(result)
     }
 
+    fn find_duplicate_resource_types(config: &Configuration) -> Vec<String>
+    {
+        let mut map: HashMap<&String, i32> = HashMap::new();
+        let mut result: HashSet<String> = HashSet::new();
+        let resource_list = &config.resources;
+        if resource_list.is_empty() {
+            return Vec::new();
+        }
+
+        for r in resource_list
+        {
+            let v = map.entry(&r.resource_type).or_insert(0);
+            *v += 1;
+            if *v > 1 {
+                result.insert(r.resource_type.clone());
+            }
+        }
+
+        result.into_iter().collect()
+    }
+
+    pub fn invoke_export(&self, _error_action: ErrorAction, _progress_callback: impl Fn() + 'static) -> Result<ConfigurationExportResult, DscError> {
+        let (config, messages, had_errors) = self.validate_config()?;
+
+        let duplicates = Self::find_duplicate_resource_types(&config);
+        if !duplicates.is_empty()
+        {
+            let duplicate = &duplicates[0];
+            return Err(DscError::Validation(format!("Resource {duplicate} specified multiple times")));
+        }
+
+        let mut result = ConfigurationExportResult {
+            result: None,
+            messages,
+            had_errors
+        };
+
+        if had_errors {
+            return Ok(result);
+        };
+        let mut conf = config_doc::Configuration::new();
+
+        for resource in &config.resources {
+            let Some(dsc_resource) = self.discovery.find_resource(&resource.resource_type).next() else {
+                return Err(DscError::ResourceNotFound(resource.resource_type.clone()));
+            };
+            let export_result = dsc_resource.export()?;
+
+            for (i, instance) in export_result.actual_state.iter().enumerate()
+            {
+                let mut r = config_doc::Resource::new();
+                r.resource_type = dsc_resource.type_name.clone();
+                r.name = format!("{}-{i}", r.resource_type);
+                let props: HashMap<String, serde_json::Value> = serde_json::from_value(instance.clone()).unwrap();
+                r.properties = Some(props);
+
+                conf.resources.push(r);
+            }
```

Comment:
This code seems very similar or same as above.  Maybe have a helper that creates a new config based on export_result?

Reply:
Updated.